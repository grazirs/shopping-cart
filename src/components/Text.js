import styled, { css } from "styled-components";
import theme from "../theme";

export const Text = styled("h1").attrs(({ level }) => ({
  as: `h${level}`,
}))`
  ${({color, fontSize, fontWeight, align}) => css`
    font-size: ${theme.fontSizes[fontSize]};
    color: ${theme.colors[color]};
    font-weight: ${theme.fontWeights[fontWeight]};
    text-align: ${align};
  `}
`;
