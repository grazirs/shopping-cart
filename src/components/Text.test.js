import renderer from 'react-test-renderer';
import { Text } from "./Text";
import theme from '../theme';
import { ThemeProvider } from 'styled-components';

describe("<Text/>", () => {
  it("renders Text component correctly", () => {
    const text = renderer.create(<ThemeProvider theme={theme}><Text /></ThemeProvider>).toJSON();
    expect(text).toMatchSnapshot();
  })
});
