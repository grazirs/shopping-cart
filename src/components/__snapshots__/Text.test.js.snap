// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> renders Text component correctly 1`] = `
<hundefined
  className="Text-sc-jb2lm6-0 gbiAWE"
/>
`;
